### PR TITLE
Fix the getPolicyName function for when the report or policies are not ready

### PR DESCRIPTION
**Pull request by wildan-m**
### Details

Fix concierge chat header on initial login

### Fixed Issues

$ https://github.com/Expensify/App/issues/44480
PROPOSAL: https://github.com/Expensify/App/issues/44480#issuecomment-2306158393

### Tests

1. Go to App
2. Login with a new Gmail account
3. Complete the onboarding modal requests
4. Take a look at the Concierge chat header
5. There should be only Concierge label on Concierge chat header
- [x] Verify that no errors appear in the JS console

### Offline tests

Should always online

### QA Steps

Same as test
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/32ba9433-357c-4e13-98b4-c57901b7fae9

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/e55b52b3-c851-4236-a7bd-8bb06ba9723c

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/cec4d600-8f19-4ea4-aab1-36987feee2a6

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/b141a1a4-9c3e-4abb-a511-8c030eb94095

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/bd991c2b-17fd-4396-accb-3bf108638264

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/07fa1814-8ff9-4009-82e3-2193f047588b

</details>

### CODE REVIEW

File: src/libs/ReportUtils.ts [hunk 1]
```diff
@@ -739,13 +739,17 @@ const unavailableTranslation = Localize.translateLocal('workspace.common.unavail
...
+    
+    // Ensure policies and report are fully loaded before proceeding.
+    // On initial login, they may be partially loaded, so additional checks are necessary to prevent unexpected behavior.
+    if (isEmptyObject(report) || !report.reportID || isEmptyObject(allPolicies)) {
         return noPolicyFound;
     }
-
-    if ((!allPolicies || Object.keys(allPolicies).length === 0) && !report?.policyName) {
+    
+    if (isEmptyObject(allPolicies) && !report?.policyName) {
```
Comment (wildan-m) [inline src/libs/ReportUtils.ts:749]: @allgandalf upon further consideration, I believe the second check is unreachable. My suggestion would be to eliminate it. Your thoughts?

**Comment by wildan-m** [comment]
@allgandalf sorry, need to convert back to draft, I'll need to test more with jest

**Comment by wildan-m** [comment]
@allgandalf My main solution failed the tests due to the possibility of using getPolicyName without any policies, but [my third solution](https://github.com/Expensify/App/issues/44480#issuecomment-2306158393) passed all the tests.

Is it ok to move with my 3rd solution?

<details><summary>Failed tests</summary>
1. opens a chat and load newer messages

https://github.com/Expensify/App/blob/8a35256bff4d66d0f0958fd9100e191a8fda404d/tests/ui/PaginationTest.tsx#L312


2. PolicyExpenseChat > Active > as member


https://github.com/Expensify/App/blob/8a35256bff4d66d0f0958fd9100e191a8fda404d/tests/unit/ReportUtilsTest.ts#L208


3. PolicyExpenseChat > Active > as member

https://github.com/Expensify/App/blob/8a35256bff4d66d0f0958fd9100e191a8fda404d/tests/unit/ReportUtilsTest.ts#L246


4. renders the policy deleted archive reason as the preview message of the chat

https://github.com/Expensify/App/blob/8a35256bff4d66d0f0958fd9100e191a8fda404d/tests/unit/SidebarTest.ts#L93

</details>

File: src/libs/ReportUtils.ts [hunk 2]
```diff
@@ -739,13 +739,17 @@ const unavailableTranslation = Localize.translateLocal('workspace.common.unavail
...
+    // Ensure policies and report are fully loaded before proceeding.
+    // On initial login, they may be partially loaded, so additional checks are necessary to prevent unexpected behavior.
+    if (isEmptyObject(report) || !report.reportID || isEmptyObject(allPolicies)) {
         return noPolicyFound;
     }
 
-    if ((!allPolicies || Object.keys(allPolicies).length === 0) && !report?.policyName) {
+    if (isEmptyObject(allPolicies) && !report?.policyName) {
         return unavailableTranslation;
     }
```
[the remark this answers is not in the document]
Comment (wildan-m) [reply src/libs/ReportUtils.ts:751]: Yeah, [I'm aware](https://github.com/Expensify/App/pull/48621/files#r1744911958) of that. This is [my latest suggestion](https://github.com/Expensify/App/pull/48621#issuecomment-2331102264).

@allgandalf what is your opinion?

**Comment by wildan-m** [comment]
@allgandalf The PR has been updated

**Comment by wildan-m** [comment]
@allgandalf sure, the videos has been updated

**Review by allgandalf — APPROVED** [review]
Tests well and LGTM

**Review by mjasikowski — APPROVED** [review]

**Pull request merged**